Source/JavaScriptCore/runtime/ObjectConstructor.cpp

```diff
@@ -387,7 +387,9 @@ JSC_DEFINE_HOST_FUNCTION(objectConstructorAssign, (JSGlobalObject* globalObject,
                 RETURN_IF_EXCEPTION(scope, { });
             }
 
-            if (objectAssignFast(vm, target, source, properties, values))
+            bool objectAssignFastSucceeded = objectAssignFast(globalObject, target, source, properties, values);
+            RETURN_IF_EXCEPTION(scope, { });
+            if (objectAssignFastSucceeded)
```

Comment:
Why do we need this change?

Reply:
In `JSObject::copyArrayStorage`, `SparseArrayValueMap:: putEntry` might throw exception. 
```
                map->putEntry(globalObject, this, it->key, value, false);
                RETURN_IF_EXCEPTION(scope, void());
```

Reply:
Updated. Now `putDirectIndex` may calls `putByIndexBeyondVectorLengthWithoutAttributes` which might throw exceptions.